src/tls/openssl/tls.c

```diff
@@ -936,6 +954,120 @@ void tls_set_verify_client(struct tls *tls)
 }
 
 
+/**
+ * Set TLS server context to request certificate from peer
+ * and set trust all certificates of peer.
+ *
+ * @deprecated Use tls_set_verify_peer_trust_all instead
+ * @param tls    TLS Context
+ */
+void tls_set_verify_client(struct tls *tls)
+{
+	if (!tls)
+		return;
+
+	tls_set_verify_client_trust_all(tls);
+}
+
+
+static int tls_verify_handler_ud(int ok, X509_STORE_CTX *ctx)
+{
+	int ret = ok;
+	struct tls_conn_d *d;
+	SSL *ssl;
+	int err, depth;
+
+	err = X509_STORE_CTX_get_error(ctx);
+
+#if (DEBUG_LEVEL >= 6)
+	char    buf[128];
+	X509   *err_cert;
+
+	err_cert = X509_STORE_CTX_get_current_cert(ctx);
+
+	X509_NAME_oneline(X509_get_subject_name(err_cert), buf, 128);
+	DEBUG_INFO("%s: subject_name = %s\n", __func__, buf);
+
+	X509_NAME_oneline(X509_get_issuer_name(err_cert), buf, 128);
+	DEBUG_INFO("%s: issuer_name  = %s\n", __func__, buf);
+#endif
+	if (err) {
+		depth = X509_STORE_CTX_get_error_depth(ctx);
+		DEBUG_WARNING("%s: err          = %d\n", __func__, err);
+		DEBUG_WARNING("%s: error_string = %s\n", __func__,
+				X509_verify_cert_error_string(err));
+		DEBUG_WARNING("%s: depth        = %d\n", __func__, depth);
+	}
+
+#if (DEBUG_LEVEL >= 6)
+	DEBUG_INFO("tls_verify_handler ok = %d\n", ok);
+#endif
+
+	ssl = X509_STORE_CTX_get_ex_data(ctx,
+		SSL_get_ex_data_X509_STORE_CTX_idx());
+
+	if (!ssl) {
+		DEBUG_WARNING("X509_STORE_CTX_get_ex_data (SSL*) failed\n");
+		return ret;
+	}
+
+	d = SSL_get_ex_data(ssl, tls_verify_idx);
+	if (!d) {
+		DEBUG_WARNING("SSL_get_app_data (struct tls_conn_d) failed\n");
+		return ret;
+	}
+
+	if (d->verifyh)
+		ret = d->verifyh(ok, d->verify_d);
+
+	return ret;
+}
+
+
+/**
+ * Enable request certificate from peer in TLS server connection
+ * Set verify handler.
+ *
+ * @param tc     TLS connection
+ * @param depth  Max depth certificate chain accepted.
+ *               A negative depth uses default depth.
+ * @param cb     SSL verify handler. If NULL default verify handler is used.
+ */
+int tls_set_verify_client_handler(struct tls_conn *tc, int depth,
+	int (*cb) (int ok, void *d), void *d)
+{
+#if !defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
I just upgraded LibreSSL for CI tests, can you try to enable these parts. At first quick look all required APIs are available now.